source/m2mtlvserializer.cpp

```diff
@@ -164,9 +177,44 @@ bool M2MTLVSerializer::serialize_multiple_resource(const M2MResource *resource,
     return success;
 }
 
+/* See, OMA-TS-LightweightM2M-V1_0-20170208-A, Appendix C,
+ * Data Types, Integer, TLV Format 
+ * 
+ * This is easy to change to support 8, 16 or 32 interger. */
+template <typename T> void M2MTLVSerializer::set_value_int(T value, uint8_t *buffer, uint32_t &size)
```

Comment:
Size is bound type of value.  This is not mandatory parameter an could handle different way because   parameter value size it known before execution of set_value_int.  This for avoiding magic numbers. Now sizeof() return always correct size and it has been called only in one place.